src/components/Entry/EntryListFilters/AIGeneratedFilter/index.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { FormEvent, useEffect, useState } from 'react';
+
+import { createSelector } from 'reselect';
+import { format } from 'url';
+
+import NumberComponent from 'components/NumberComponent';
+import Tooltip from 'components/SimpleCommonComponents/Tooltip';
+import { getPayloadOrEmpty } from 'components/FiltersPanel';
+
+import loadData from 'higherOrder/loadData/ts';
+import descriptionToPath from 'utils/processDescription/descriptionToPath';
+
+import { goToCustomLocation } from 'actions/creators';
+
+import cssBinder from 'styles/cssBinder';
+
+import style from 'components/FiltersPanel/style.css';
+import stylego from 'pages/style.css';
+
+const css = cssBinder(stylego, style);
+
+const categories = {
+  Curated: 'curated',
+  'AI-Generated (reviewed)': 'ai-reviewed',
+  'AI-Generated (unreviewed)': 'ai-unreviewed',
+};
+type Categories = keyof typeof categories;
+
+type Props = {
+  label?: string;
+  changeLabel: (s: string) => void;
+  goToCustomLocation?: typeof goToCustomLocation;
+  customLocation?: InterProLocation;
+};
+
+interface LoadedProps extends Props, LoadDataProps<GroupByPayload> {}
+
+const AIGeneratedFilter = ({
+  changeLabel,
+  data,
+  isStale,
+  customLocation,
+  goToCustomLocation,
+}: LoadedProps) => {
+  if (!data || !customLocation) return null;
+  const { loading, payload } = data;
+  const { search } = customLocation;
+
+  const _handleSelection = ({ target }: FormEvent) => {
+    const value = (target as HTMLInputElement).value;
+    const { page, curation_status: _, cursor: __, ...restOfsearch } = search;
+    if (value !== 'All') restOfsearch.curation_status = value;
+    goToCustomLocation?.({ ...customLocation, search: restOfsearch });
+  };
+
+  const terms = Object.entries<number>(
+    getPayloadOrEmpty(payload, loading, isStale),
+  ).sort(([, a], [, b]) => b - a);
+
+  if (!loading) {
+    terms.unshift(['All', NaN]);
+  }
+
+  const [showFilter, setShowFilter] = useState(true);
+
+  useEffect(() => {
+    if (terms.length > 0) {
+      // If loaded AI-generated reviewed and unreviewed counts will always be in position 2 and 3 of the terms obj
+      if (terms[2][1] == 0 && terms[3][1] == 0) {
```

Comment:
Are you sure about that?
`terms[0]` is always `['All', NaN]`, but other items are sorted by number of entries/signatures in the category. It's true that _at the moment_ there are always more curated entries than AI-generated, but you should not assume that it always be the case. 

You should check using the term, e.g.

```ts
if (terms.some(([term, count]) => (term === "AI-Generated (reviewed)" || term === "AI-Generated (unreviewed)") && count > 0)) {
  setShowFilter(true);
} else {
  setShowFilter(false);
}
```

Reply:
Missed the sorting part. Makes sense now. I'll fix this.